client/js/s3/uploader.basic.js

```diff
@@ -157,7 +158,19 @@
         },
 
         setCredentials: function(credentials, ignoreEmpty) {
-            if (credentials && credentials.secretKey) {
+            if (credentials && ignoreEmpty) {
+                if (this._currentCredentials === undefined) {
+                    this._currentCredentials = {};
+                }
+
+                credentials.accessKey !== undefined && (this._currentCredentials.accessKey = credentials.accessKey);
```

Comment:
These are a bit too dense and hard to follow, especially the 3rd line. These can probably be expanded into `if` statements and/or moved into other functions.